src/peft/tuners/boft/layer.py

```diff
@@ -777,7 +770,7 @@ def update_layer(
         self.boft_R[adapter_name] = nn.Parameter(
             torch.zeros(boft_n_butterfly_factor + 1, boft_block_num, boft_block_size, boft_block_size)
         )
-        self.boft_s[adapter_name] = nn.Parameter(torch.ones(1, int(self.out_features)))
+        self.boft_s[adapter_name] = nn.Parameter(torch.ones(int(self.out_features), 1))
```

Comment:
So this transposes the `boft_s` vector. Could you explain why that is important? I was wondering if this would prevent the state dict from being loaded successfully but it seems it's not an issue (see my longer comment).